screencaptor/src/main/java/com/wealthfront/screencaptor/ScreenCaptor.kt

```diff
@@ -85,199 +84,59 @@ object ScreenCaptor {
-   * So remember to pass in a valid path retrieved from the context as follows
-   * {@code context.filesDir or context.getExternalFilesDir(Environment.DIRECTORY_PICTURES)}.
-   *
-   * @param screenshotFormat specifies the format of the screenshot file.
-   *
-   * @param screenshotQuality specifies the level of compression of the screenshot.
-   *
-   */
-  @Synchronized
-  fun takeScreenshot(
-    views: List<View>,
-    screenshotName: String,
-    screenshotNameSuffix: String = "",
-    viewIdsToExclude: Set<Int> = setOf(),
-    viewModifiers: Set<DataModifier> = setOf(),
-    viewDataProcessor: ViewDataProcessor = DefaultViewDataProcessor(),
-    viewMutators: Set<ViewMutator> = setOf(ScrollbarHider, CursorHider),
-    screenshotDirectory: String = defaultScreenshotDirectory,
-    screenshotFormat: ScreenshotFormat = PNG,
-    screenshotQuality: ScreenshotQuality = BEST
-  ): String {
-    views.forEach { view ->
-      Log.d(SCREENSHOT, "Root view has height(${view.height}) and width(${view.width})")
-      if (view.width == 0 || view.height == 0) {
-        throw IllegalStateException("This view ($view) has no height or width. Is ${view.id} the currently displayed activity?")
-      }
-    }
-
+  ) {
     if (!File(screenshotDirectory).exists()) {
       Log.d(SCREENSHOT, "Creating directory $screenshotDirectory since it does not exist")
       val screenshotDirsCreated = File(screenshotDirectory).mkdirs()
       assert(screenshotDirsCreated)
     }
 
-    val bitmap = createCanvasBitmap(views)
-    val canvas = Canvas(bitmap)
-
-    val viewsToCapture = AtomicInteger(views.size)
     val deviceName = MANUFACTURER.replaceWithUnderscore() + "_" + MODEL.replaceWithUnderscore()
-    val screenshotId = "${screenshotName.toLowerCase(ENGLISH)}_${deviceName}_${SDK_INT}_$screenshotNameSuffix"
-    val screenshotFilePath = "$screenshotDirectory/$screenshotId.${screenshotFormat.extension}"
+    val screenshotId =
+      "${screenshotName.toLowerCase(ENGLISH)}_${deviceName}_${SDK_INT}_$screenshotNameSuffix"
+    val screenshotFile = File("$screenshotDirectory/$screenshotId.${screenshotFormat.extension}")
 
-    mainHandler.post {
+    val initialStateOfViews = hashMapOf<View, ViewTreeState>()
+    getRootViewsFromActivity(activity).forEach { rootView ->
       Log.d(SCREENSHOT, "Modifying view tree for '$screenshotName'")
-      val initialStateOfViews = hashMapOf<View, ViewTreeState>()
-      views.forEach { rootView ->
-        val viewState = modifyViewBeforeScreenshot(
-          rootView,
-          viewDataProcessor,
-          viewMutators,
-          viewModifiers,
-          viewIdsToExclude
-        )
-        initialStateOfViews[rootView] = viewState
+      if (rootView.width == 0 || rootView.height == 0) {
+        throw IllegalStateException("This view ($rootView) has no height or width. Is ${rootView.id} the currently displayed activity?")
       }
 
-      views.forEach { rootView ->
-        rootView.post {
```

Comment:
hmm do we still need per-view posting: https://github.com/wealthfront/screencaptor/pull/8/files